plugins/out_s3/s3.c

```diff
@@ -137,7 +141,7 @@ static int create_headers(struct flb_s3 *ctx, struct flb_aws_header **headers, i
         s3_headers[n].val_len = strlen(ctx->content_type);
         n++;
     }
-    if (ctx->compression != NULL) {
+    if (ctx->compression == COMPRESS_GZIP) {
```

Comment:
I think we should create a header for apache arrow so that users can know if the data is compressed from the log.